test/dynamics/pulse/test_pulse_to_signals.py

```diff
@@ -303,15 +320,79 @@ def test_multiple_channels_with_gaps(self):
         samples2 = np.append(np.zeros(15), np.append(gauss_samples, np.zeros(10)))
         samples3 = np.append(np.zeros(20), np.append(gauss_samples, np.zeros(5)))
 
-        self.assertAllClose(signals[0].samples, samples0, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[1].samples, samples1, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[2].samples, samples2, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[3].samples, samples3, atol=1e-14, rtol=1e-14)
+        self.assertAllClose(signals[0].samples, samples0, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[1].samples, samples1, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[2].samples, samples2, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[3].samples, samples3, atol=1e-7, rtol=1e-7)
         self.assertTrue(signals[0].carrier_freq == 5.0)
         self.assertTrue(signals[1].carrier_freq == 3.1)
         self.assertTrue(signals[2].carrier_freq == 0.0)
         self.assertTrue(signals[3].carrier_freq == 4.0)
 
+    def test_get_samples(self):
```

Comment:
Rewrite these tests to still call the `InstructionToSignals` converter. `get_samples` and `lru_cache_expr` are both internal functions, so ideally we won't directly test them. There are exceptions to this in Dynamics, but in this case I don't think it's necessary. The inputs/outputs of these functions are directly fed from/to the converter, so I don't think anything is gained by directly testing these functions, as opposed to verifying the behaviour of the converter on a symbolic pulse.

Maybe change the name of this to `test_SymbolicPulse`.

Reply:
improved and changed the name at https://github.com/Qiskit/qiskit-dynamics/pull/149/commits/e2c5148dade2da23c125e1471580e32046669d45